packages/dd-trace/src/config-helper.js

```diff
@@ -0,0 +1,90 @@
+'use strict'
+
+// TODO: Add jira tickets
+
+const { debuglog, deprecate } = require('util')
+const { supportedConfigurations, aliases, deprecations } = require('./supported-configurations.json')
+
+const aliasToCanonical = {}
+for (const canonical of Object.keys(aliases)) {
+  for (const alias of aliases[canonical]) {
+    if (aliasToCanonical[alias]) {
+      throw new Error(`The alias ${alias} is already used for ${aliasToCanonical[alias]}.`)
+    }
+    aliasToCanonical[alias] = canonical
+  }
+}
+
+// TODO: Consider to join deprecations with aliases by just making those entries an object.
+const deprecationMethods = {}
+for (const deprecation of Object.keys(deprecations)) {
+  deprecationMethods[deprecation] = deprecate(
+    () => {},
+    `The environment variable ${deprecation} is deprecated. Please use ${aliasToCanonical[deprecation]} instead.`,
+    `DATADOG_${deprecation}`
+  )
+}
+
+const debug = debuglog('dd:debug')
+
+module.exports = {
+  /**
+   * Returns the environment variables that are supported by the tracer.
+   *
+   * @returns {Partial<process.env>} The environment variables
+   */
+  getConfigurations () {
+    const configs = {}
+    for (const [env, value] of Object.entries(process.env)) {
+      if (typeof env === 'string' && (env.startsWith('DD_') || env.startsWith('OTEL_'))) {
+        if (supportedConfigurations[env]) {
+          configs[env] = value
+        } else if (aliasToCanonical[env]) {
+          // The alias should only be used if the actual configuration is not set
+          if (configs[aliasToCanonical[env]] === undefined) {
+            // In case that more than a single alias exist, use the one defined first in our own order
+            for (const alias of aliases[aliasToCanonical[env]]) {
+              if (process.env[alias] !== undefined) {
+                configs[aliasToCanonical[env]] = value
+                break
+              }
+            }
+          }
+          deprecationMethods[env]?.()
+        } else {
+          debug(
+            `Missing configuration ${env} in supported-configurations file. The environment variable is ignored.`
+          )
+        }
+      } else {
+        configs[env] = value
```

Comment:
This will also return environment variables not starting with `DD_` or `OTEL_`. That goes against what's stated in the JSDoc comment. Either that or this logic has to change

Reply:
The comment is actually somewhat correct (environment variables that don't start with DD_ or OTEL_ are generally supported). Do you have a suggestion to make the comment clearer? :)